File: test_archives/cpp_tests/batch5/test_function3.cpp
#include "../catch.hpp"
#include "../functions/function3.cpp"

TEST_CASE("Function3 initialization", "[function3]") {
    Function3 function3("John", 25, "Male", "123 Main Street", "555-555-5555");

    REQUIRE(function3.name == "John");
    REQUIRE(function3.age == 25);
    REQUIRE(function3.gender == "Male");
    REQUIRE(function3.address == "123 Main Street");
    REQUIRE(function3.phone_number == "555-555-5555");
}

TEST_CASE("Function3 calculate value", "[function3]") {
    Function3 function3("John", 25, "Male", "123 Main Street", "555-555-5555");

    REQUIRE(function3.calculate_value() == 25 + 555555555);
}

File: test_archives/py_tests/batch5/test_function3.py
from functions.function3 import function3
import pytest

@pytest.fixture
def function3():
    return function3('John', 25, 'Male', '123 Main Street', 555-555-5555)

def test_function3_init(function3):
    assert function3.name == 'John'
    assert function3.age == 25
    assert function3.gender == 'Male'
    assert function3.address == '123 Main Street'
    assert function3.phone_number == '555-555-5555'
  
def test_function3_calculate_value(function3):
    assert function3.calculate_value() == 25 + 555555555